evap/staff/importers/user.py

```diff
@@ -0,0 +1,384 @@
+        user.full_clean(validate_unique=False)
+
+
+@dataclass
+class UserInputRow(InputRow):
+    """Raw representation of a semantic user importer row, independent on the import format (xls, csv, ...)"""
+
+    column_count = 4
+
+    location: ExcelFileLocation
+
+    # Cells in the order of appearance in a row of an import file
+    title: str
+    first_name: str
+    last_name: str
+    email: str
+
+    def as_parsed_row(self):
+        user_data = UserData(
+            first_name=self.first_name,
+            last_name=self.last_name,
+            email=self.email,
+            title=self.title,
+        )
+
+        return UserParsedRow(location=self.location, user_data=user_data)
+
+
+@dataclass
+class UserParsedRow:
+    """
+    Representation of an User Row after parsing the data into the resulting data structures.
+    Basically the same, but used for interface consistency between the Importers
+    """
+
+    location: ExcelFileLocation
+
+    user_data: UserData
+
+
+class UserDataEmptyFieldsChecker(Checker):
+    """Assert email, first name and last name are not empty"""
+
+    def check_userdata(self, user_data: UserData, location: ExcelFileLocation):
+        if user_data.email == "":
+            self.importer_log.add_error(
+                _("{location}: Email address is missing.").format(location=location),
+                category=ImporterLogEntry.Category.USER,
+            )
+
+        if user_data.first_name == "":
+            self.importer_log.add_error(
+                _("{location}: User {email}: First name is missing.").format(location=location, email=user_data.email),
+                category=ImporterLogEntry.Category.USER,
+            )
+
+        if user_data.last_name == "":
+            self.importer_log.add_error(
+                _("{location}: User {email}: Last name is missing.").format(location=location, email=user_data.email),
+                category=ImporterLogEntry.Category.USER,
+            )
+
+
+class UserDataMismatchChecker(Checker):
+    """Assert UserData matches previous occurrences in the import as well as the database"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # maps user's mail to UserData instance where it was first seen to have O(1) lookup
+        self.users: Dict[str, UserData] = {}
+
+        self.in_file_mismatch_tracker: FirstLocationAndCountTracker = FirstLocationAndCountTracker()
+
+    def check_userdata(self, user_data: UserData, location: ExcelFileLocation):
+        if user_data.email == "":
+            # UserDataEmptyFieldsChecker will give an error for these, no need to spam additional errors
+            return
+
+        stored_user_data = self.users.setdefault(user_data.email, user_data)
+        if user_data != stored_user_data:
+            self.in_file_mismatch_tracker.add_location_for_key(location, user_data.email)
+
+    def finalize(self):
```

Comment:
As this method is not trivial, I suggest already today:
```suggestion
    def finalize(self) -> None:
```